src/hooks/constructionSites.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useContext, useState } from 'react';
+import { useQuery } from 'react-apollo';
+
+import { graphqlFetchPolicy, isUpcomingDate } from '../helpers';
+import { filterForValidConstructionSites } from '../jsonValidation';
+import { NetworkContext } from '../NetworkProvider';
+import { getQuery, QUERY_TYPES } from '../queries';
+import { ConstructionSite, GenericItem } from '../types';
+
+type Payload = {
+  direction?: string;
+  cause?: string;
+  restrictions?: Array<{
+    description: string;
+  }>;
+};
+
+const parseConstructionSite = (
+  constructionSite: GenericItem<Payload>
+): Partial<ConstructionSite> => {
+  let location:
+    | {
+        lat: number;
+        lon: number;
+      }
+    | undefined;
+  if (constructionSite?.locations?.[0]?.geoLocation) {
+    location = {
+      lat: constructionSite.locations?.[0].geoLocation?.latitude,
+      lon: constructionSite.locations?.[0].geoLocation?.longitude
+    };
+  }
+
+  return {
+    id: constructionSite.id,
+    startDate: constructionSite.dates?.[0]?.dateStart,
+    title: constructionSite.title,
+    category: constructionSite.categories?.[0]?.name,
+    cause: constructionSite.payload.cause,
+    description: constructionSite.contentBlocks?.[0]?.body,
+    direction: constructionSite.payload.direction,
+    endDate: constructionSite.dates?.[0]?.dateEnd,
+    imageUri: constructionSite.mediaContents?.[0]?.sourceUrl.url,
+    location,
+    locationDescription: constructionSite.locations?.[0]?.name,
+    restrictions: constructionSite.payload.restrictions?.map((value) => value.description)
+  };
+};
```

Comment:
could this better be part of `queryDataParser` helper to reduce the code to the hook?

Reply:
i did not like putting them into the queryDataParser file as well, because that is already rather big.

instead i moved the queryDataParser file to a new subdirectory `parser` and renamed it to `listItemParser` and added a new file for the construction site parsing.